bindings/matrix-sdk-ffi/src/room_list.rs

```diff
@@ -584,16 +584,32 @@ impl RoomListItem {
     }
 
     /// Builds a `Room` FFI from an invited room without initializing its
-    /// internal timeline
+    /// internal timeline.
     ///
-    /// An error will be returned if the room is a state different than invited
+    /// An error will be returned if the room is a state different than invited.
     ///
     /// ⚠️ Holding on to this room instance after it has been joined is not
-    /// safe. Use `full_room` instead
+    /// safe. Use `full_room` instead.
+    #[deprecated(note = "Please use `room_without_timeline` instead.")]
     fn invited_room(&self) -> Result<Arc<Room>, RoomListError> {
-        if !matches!(self.membership(), Membership::Invited) {
+        self.room_without_timeline()
+    }
+
+    /// Builds a `Room` FFI from a room without initializing its internal
+    /// timeline.
+    ///
+    /// An error will be returned if the room is a state other than invited
+    /// or knocked, the 2 states that would match this use case.
+    ///
+    /// ⚠️ Holding on to this room instance after it has been joined is not
+    /// safe. Use `full_room` instead.
+    fn room_without_timeline(&self) -> Result<Arc<Room>, RoomListError> {
+        let membership = self.membership();
+        if !matches!(membership, Membership::Invited)
+            && !matches!(self.membership(), Membership::Knocked)
```

Comment:
It can be a single `matches!`.

```suggestion
        if !matches(membership, Membership::Invited | Membership::Knocked)
```